### PR TITLE
Allow querying objects by type

### PR DESCRIPTION
**Pull request by dancsi**

### CODE REVIEW

**Comment by Lapin0t** [comment]
This endpoint returns a collection of resources so it looks like a to-many relationship. Too keep things consistent in a jsonapi-like fashion, shouldn't we only include the id+href to the resources, ie return an array of linkage objects? This by-type endpoint is probably gonna return a lot of stuff. Note that we could add an opt-in to include the linked objects with `/store/by-type/{type}?include`.

**Comment by flupe** [comment]
On second thought, this does feel more consistent.

**Comment by dancsi** [comment]
In that case, I guess it would be nice to add a `?depth=n` parameter to both this and relationships? What about pagination, while we are at it?

**Comment by Lapin0t** [comment]
What would be the semantic for `?depth=n`? (i'm realizing i didn't specify semantics for `?include` either: it would include every resource of the array in an `included` object at the top-level of the response).

On pagination: we talked about it with @flupe this evening and the conclusion was along the line of: we want it but it's maybe not the biggest priority.

**Comment by dancsi** [comment]
`?depth=n` is just a way to resolve the relationships `n` levels in depth. That may or may not be a good idea, as it might lead to denial of service, as it would allow people to query the database for eeeeverything...

**Comment by flupe** [comment]
> `?depth=n` is just a way to resolve the relationships n levels in depth. That may or may not be a good idea, as it might lead to denial of service, as it would allow people to query the database for eeeeverything...
Well this was initially how I thought about including objects, but I think we would be better off requiring people to be specific about what to include.
the `?depth` parameter is and all-or-nothing param, there is no fine-tuning, and with `to-many` relationships, having a depth of more than 1 will quickly lead to huge data exchange.
We still need to define exactly how `?include` will work, but if we stay consistent with the JSON API, it will be quite easy to include specific nested resources, this just requires to be explicit about it.
Example: If a resource of type `article` has a relationships named comments, you could do `?include=comments.author`.
So in conclusion, I think having will `?include` will do just fine.

**Comment by dancsi** [comment]
I agree, the only question is whether we merge this first, or we wait for an implementation of  `?include=stuff`. Its semantics are defined relatively well in jsonapi already.
